examples/wsmi.py

```diff
@@ -0,0 +1,455 @@
+ax.set_xlabel("Parameter Combination (kernel_tau)")
+ax.set_title("Effect of kernel and tau Parameters on wSMI")
+ax.grid(True, alpha=0.3)
+
+# Add value labels on bars
+for bar, value in zip(bars, values):
+    height = bar.get_height()
+    ax.text(
+        bar.get_x() + bar.get_width() / 2.0,
+        height + 0.001,
+        f"{value:.3f}",
+        ha="center",
+        va="bottom",
+    )
+
+plt.tight_layout()
+plt.show()
+
+# %%
+# Comparing wSMI vs SMI (weighted vs unweighted)
+# ===============================================
+#
+# wSMI uses distance-based weighting of symbolic patterns, while standard SMI
+# treats all patterns equally. Let's compare their sensitivity.
+
+# Compute both weighted and unweighted versions
+conn_wsmi = wsmi(epochs, kernel=3, tau=1, weighted=True, verbose=False)
+conn_smi = wsmi(epochs, kernel=3, tau=1, weighted=False, verbose=False)
+
+# Extract connectivity values (Source ↔ Linear Coupled)
+wsmi_values = conn_wsmi.get_data().mean(axis=0)[0]
+smi_values = conn_smi.get_data().mean(axis=0)[0]
+
+print("\nComparison of wSMI vs SMI:")
+print(f"wSMI (weighted):     {wsmi_values:.3f}")
+print(f"SMI (unweighted):    {smi_values:.3f}")
+print(f"Improvement ratio:   {wsmi_values / smi_values:.2f}x")
+
+# %%
+# Anti-aliasing: Understanding the Preprocessing
+# ==============================================
+#
+# wSMI includes automatic anti-aliasing filtering to prevent artifacts when
+# ``tau > 1``. Let's demonstrate the difference this makes.
+
+
+# Compute with and without anti-aliasing
+conn_filtered = wsmi(epochs, kernel=3, tau=2, anti_aliasing=True, verbose=False)
+
+with warnings.catch_warnings(record=True) as w:
+    warnings.simplefilter("always")
+    conn_unfiltered = wsmi(epochs, kernel=3, tau=2, anti_aliasing=False, verbose=False)
+    if w:
+        print(f"Warning when anti-aliasing disabled: {w[0].message}")
+
+# Compare results
+filtered_connectivity = conn_filtered.get_data().mean(axis=0)[0]
+unfiltered_connectivity = conn_unfiltered.get_data().mean(axis=0)[0]
+
+print("\nAnti-aliasing Effect (tau=2):")
+print(f"With anti-aliasing:    {filtered_connectivity:.3f}")
+print(f"Without anti-aliasing: {unfiltered_connectivity:.3f}")
+print(
+    f"Difference:            {abs(filtered_connectivity - unfiltered_connectivity):.3f}"
+)
+
+# %%
+# Using Real EEG Data: Sample Dataset
+# ===================================
+#
+# Now let's apply wSMI to real EEG data from the MNE sample dataset.
+
+# Load sample data
+data_path = sample.data_path()
+raw_fname = data_path / "MEG" / "sample" / "sample_audvis_filt-0-40_raw.fif"
+event_fname = data_path / "MEG" / "sample" / "sample_audvis_filt-0-40_raw-eve.fif"
+
+# Load and preprocess data
+raw = mne.io.read_raw_fif(raw_fname, verbose=False)
+events = mne.read_events(event_fname, verbose=False)
+
+# Pick a few EEG channels for demonstration
+eeg_picks = ["EEG 001", "EEG 002", "EEG 003", "EEG 004"]
+raw.pick_channels(eeg_picks)
```

Comment:
Something I just noticed: `pick_channels()` is deprecated so let's use the recommended `pick()`.

```suggestion
raw.pick(eeg_picks)
```